### ISSUE TITLE
Display Overbuy Rule to Game Players

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#3767]
Task here is to add a display/label/notification of how over-buy is handled in a given game. Similar perhaps to game help notes, it'll be something you can click (or maybe a help button on purchase window), to then read about how overbuy is handled, whether it is refunded, carried over, or lost.

Currently a user would have to experiment with this or find it out from the map XML. The feedback and notification of which rule is being used is missing from the UI.

**Comment by panther2** [comment]
While there the correct handling of overbuys should be tested and corrected 
In v4 for example a refund for an unintentional overbuy should be given, the overbuys are lost without refund instead.
In Global 1940 a refund for an unintentional overbuy should be given, the overbuys are carried over to the next turn(s) instead.
Not to repeat the discussion of #3506 here, but the implementation of the handling apperas to be wrong on some wwII-maps. Just a reminder here - when I find some time, I will test it wwII-map by map and open separate issues.

**Comment by DanVanAtta** [comment]
Good points @panther2 , for each of those problems we should create a map bug. If/when we fix this issue, it'll be easier to identify those maps, we'll see what the refund policy is and it'll be easier to know if it's wrong compared to testing the hard way which refund policies are used on which maps.

**Comment by Cernelius** [comment]
Good idea, but this topic should be reformulated on more general terms, I believe, as this property/rule is just "one of".

For example, in 270BC people usually get surprised that city don't block blitz, as all rulesets from v2 onwards do.

I guess it would be good having something after unit help in which to see all the properties that are currently set "true" for that game, like power users can already do by delving into the xml.

Tho there are a lot of properties that are or could be documented in the unit help info; for example, if a factory blocks blitz or not could be clarified there. So maybe those should not be listed somewhere else too, or it might be redundant. On the other hand, in the moment you cannot decide if this or that capturable unit blocks blitz, but you can only set for them all the same, then the redundancy may be to have it part of the specific units info.

The thing can stay in "Help", in between of "Unit Help" and "Game Notes".

Alternatively, a button in Game/Map Options to show all the properties, not just the editable ones.

p.s.: "Map Options" should be named "Game Options", or just "Options", since it is part of the "Game" submenu.

p.p.s.: On the other hand, showing up all the properties may be not polished enough, for regular users, especially in cases like WWIIv3, in which you would, then, see stuff like:
`property name="neutralCharge" value="9999999"`
That I don't think would be so helpful.
That is there just because lacking a way to define infinite (one of the several cases, in which the only solution is to write down a very big number).
But also all the cases in which we set "-1" to mean infinite are not really good candidates to be displayed to regular users (tho this is already happening if they are editable).

I'm just saying that I suggest reformulating this issue as to generally display all relevant properties to the users (comprising over-buy), but also pointing out that just showing up all properties as they are is not very polished or really informative at all in all cases.

**Comment by DanVanAtta** [comment]
> I'm just saying that I suggest reformulating this issue as to generally display all relevant properties to the users (comprising over-buy), but also pointing out that just showing up all properties as they are is not very polished or really informative at all in all cases.

I don't think that will be effective, it also increase the scope of this task. A list of yes/no votes would be okay, but this is not the issue/task to do that. This task is specific and actionable, the scope is well defined. We should not muddy that up; any further discussion of additional tasks should go elsewhere. Each of those rules are complicated enough that in code we'll have to do each one carefully each on their own. Discussing all of them at once is not going to help that and will not get us very fast, instead cut many issues and we'll close them down quickly as we knock out the task or decide we will not do it. We do not want any non-discussion tasks running for longer than a 4 weeks, tracking tasks are meant to be short and sweet, background context ideally behind a link unless the topic needed lots of discussion for requirements. 

> For example, in 270BC people usually get surprised that city don't block blitz, as all rulesets from v2 onwards do.

Please create a new ticket to report this player confusion, this is a UX problem. The fix here is not at all on labeling how the game handles purchase that cannot be placed.

> Tho there are a lot of properties that are or could be documented in the unit help info;

Yup, that is in unit help though, please create a ticket for that and the properties you would like to see documented in that window.

> . On the other hand, in the moment you cannot decide if this or that capturable unit blocks blitz, but you can only set for them all the same, then the redundancy may be to have it part of the specific units info.

My recommendation is for the unit icon to try and help solve that on a map by map basis, give a special symbol for blitz. There could be maybe a game engine feature here. This is a better topic for forum though.

> The thing can stay in "Help", in between of "Unit Help" and "Game Notes".

.... this has nothing to do with how the game handles purchase, re-reading this topic will take time and reading items off-topic is not helpful for identifying what requirements we need to support and then deciding how to update the code to do that.

> Alternatively, a button in Game/Map Options to show all the properties, not just the editable ones.

I think that is a very good idea, but that would also deserve it's own feature-request/forum post.

> p.s.: "Map Options" should be named "Game Options", or just "Options", since it is part of the "Game" submenu.

Another forum post or github issue that should be created

> p.p.s.: On the other hand, showing up all the properties may be not polished enough, for regular users, especially in cases like WWIIv3, in which you would, then, see stuff like:
property name="neutralCharge" value="9999999"
That I don't think would be so helpful.

This is crossing threads, https://forums.triplea-game.org/topic/912/a-general-way-to-define-infinite/15

Please keep the topic in one place to avoid confusion. In general, please prefer short and on-topic issues that are focused.

**Comment by Cernelius** [comment]
Ok if you prefer keeping it focused, that's your call. Just to make sure there is no miscommunication, what I was saying is that a power user currently can get the info by opening the xml and looking if there is this property set true:

    <property name="Unplaced units live when not placed" value="true" editable="false">
      <boolean/>
    </property>

That is now currently visible in Map Options only if editable.

So this issue could be solved by letting regular users see all the properties (thus comprising this particular one), without having to open the xml. Not saying this is the best way to solve this specific case.

**Comment by DanVanAtta** [comment]
Marking as high impact:
- confusion on buy rules could result in an undesirable purchase to only find out at the end of round. This can be an unexpected defeat due to unclear signaling of game rules.

**Comment by panther2** [comment]
After discussions in various issues, here is a summary of the rules over versions:
v1: units are lost, no refund
v2: units are lost, no refund
(per FAQ: Technically (which means, in a tournament), yes. In a friendly game, tell the player he's making a mistake and let him take it back or refund him the money. You don't want to win that way.)
v2: LHTR: units are lost, no refund
v3: units are not lost but can be placed during a future mobilization phase
v4; return units, get refund (only in case it occured inadvertently)
v5: return units, get refund (only in case it occured inadvertently)
v6: return units, get refund (only in case it occured inadvertently)
Europe, Pacific, Global 1940: return units, get refund (only in case it occured inadvertently)

In other words: The rule for v4+ is "Overbuy is forbidden, there is no refund (except "unintentional").

**Comment by Cernelius** [comment]
If you can confirm this, I believe the "only in case it occurred inadvertently" is actually a rule common to all rulesets, from v1 to v6 and comprising EPG. Meaning you are not allowed to willingly purchase more units than you can possibly place in v1 and v2, nor in v3 (especially not in v3, since you would keep them).

Also, I would not say that the refund is granted only in case it occurred inadvertently, as that sounds like say I can willingly overpurchase, and just losing the refund, while I believe this is not an option.

The other side of the coin is that you are also not allowed to end the Place/Mobilize Units phase if you have unplaced units that you can still place (but I don't recall if such rules existed already in v1 and v2).

**Comment by ron-murhammer** [comment]
@DanVanAtta This is a good example of how labels have really become meaningless. We have an issue marked as problem and feature request which to me doesn't really make any sense.

**Comment by DanVanAtta** [comment]
@ron-murhammer not really on-topic here. Context matters, we used to use the feature request label. Not displaying over-buy in any way is a problem as you don't know how the game handles it. I've been hit with this thinking the game would refund to see units lost or queued. Not placing all units that you can is game changing.

**Comment by panther2** [comment]
> @DanVanAtta This is a good example of how labels have really become meaningless. We have an issue marked as problem and feature request which to me doesn't really make any sense.

@ron-murhammer 

What I understand is that "Display Overbuy Rule to Game Players" started as a feature request. In the end it turned out that there is not only a task of "displaying the rule" but a task of correcting the  way TripleA handles overbuys. 
So I don't think it is a labelling problem, but a question of either adjusting the initial headline or adding another issue.
